api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java

```diff
@@ -245,22 +247,28 @@ private RdrWorkspace toRdrWorkspace(DbWorkspace dbWorkspace) {
     rdrWorkspace.setIntendToStudy(dbWorkspace.getIntendedStudy());
     rdrWorkspace.setFindingsFromStudy(dbWorkspace.getAnticipatedFindings());
 
-    // Call Firecloud to get a list of Collaborators
-    FirecloudWorkspaceACL firecloudResponse =
-        fireCloudService.getWorkspaceAcl(
-            dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName());
-    Map<String, FirecloudWorkspaceAccessEntry> aclMap = firecloudResponse.getAcl();
+    try {
+      // Call Firecloud to get a list of Collaborators
+      List<UserRole> collaboratorsMap =
+          workspacesApi.getFirecloudUserRoles(
+              dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName());
 
-    // Since the USERS cannot be deleted from workbench yet, hence sending the the status of
-    // COLLABORATOR as ACTIVE
-    aclMap.forEach(
-        (email, access) -> {
-          RdrWorkspaceUser workspaceUserMap = new RdrWorkspaceUser();
-          workspaceUserMap.setUserId((int) userDao.findUserByUsername(email).getUserId());
-          workspaceUserMap.setRole(RdrWorkspaceUser.RoleEnum.fromValue(access.getAccessLevel()));
-          workspaceUserMap.setStatus(RdrWorkspaceUser.StatusEnum.ACTIVE);
-          rdrWorkspace.addWorkspaceUsersItem(workspaceUserMap);
-        });
+      // Since the USERS cannot be deleted from workbench yet, hence sending the the status of
+      // COLLABORATOR as ACTIVE
+      collaboratorsMap.forEach(
+          (userRole) -> {
+            RdrWorkspaceUser workspaceUserMap = new RdrWorkspaceUser();
+            workspaceUserMap.setUserId(
+                (int) userDao.findUserByUsername(userRole.getEmail()).getUserId());
+            workspaceUserMap.setRole(
+                RdrWorkspaceUser.RoleEnum.fromValue(userRole.getRole().toString()));
+            workspaceUserMap.setStatus(RdrWorkspaceUser.StatusEnum.ACTIVE);
+            rdrWorkspace.addWorkspaceUsersItem(workspaceUserMap);
+          });
+    } catch (Exception ex) {
+      log.warning("Exception while retrieving workspace Collaborators");
+      rdrWorkspace.addWorkspaceUsersItem(new RdrWorkspaceUser());
```

Comment:
Will the RDR really be happy to receive a totally-empty entry for workspace users? I might have expected here to either: (1) skip exporting this workspace entirely, since the collaborators list is pretty important to the state of the workspace, or (2) leave the collaborators list empty.

Reply:
As discussed offline ,in case of error we will not log the workspace id for which there was an error and will skip it